server/src/main/java/com/google/fhir/gateway/interfaces/RequestMutation.java

```diff
@@ -32,4 +33,8 @@ public class RequestMutation {
   // parameter in RequestMutation is already present in the original request).
   // Old parameter values should be explicitly retained while mutating values for that parameter.
   @Builder.Default Map<String, List<String>> queryParams = new HashMap<>();
```

Comment:
nit: Consider changing this to `extraQueryParams` or `additionalQueryParams` to make its purpose more clear (not a blocker as it is not directly added in this PR).

Reply:
Done